src/sshkeys/sshkeys.controller.ts

```diff
@@ -0,0 +1,59 @@
+import { Body, Controller, Delete, Get, HttpException, HttpStatus, Param, Post, Query, Req } from '@nestjs/common';
+import { SetSshKeyDto } from './dto/set-sshkey.dto';
+import { SshKeysService } from './sshkeys.service';
+
+@Controller('sshkeys')
+export class SshKeysController {
+    constructor(
+        private sshkeysService: SshKeysService
+    ) {}
+
+    // PATCH /sshkeys/:username
+    // This action adds a key ${setSshDto} to a user's keys
+    @Post(':username')
+    async setSshKey(@Param('username') username: string, @Body() setSshDto: SetSshKeyDto, @Req() req: Request) {
+        try {
+            if (username !== req['user']?.username) {
```

Comment:
As a result of the above change, you can just use `user` instead of `req['user']`:
```suggestion
            if (username !== user?.username) {
```